### PR TITLE
Allow render in controllers to accept the view

### PR DESCRIPTION
**Pull request by scrogson**
We should support `render(conn, View, "template")` and `render(conn, View, "template", assigns)`.

### CODE REVIEW

File: lib/phoenix/controller.ex [hunk 1]
```diff
@@ -468,26 +472,41 @@ defmodule Phoenix.Controller do
...
+  @spec render(Plug.Conn.t, atom, atom | binary, Dict.t) :: Plug.Conn.t
+  def render(conn, view, template, assigns)
+    when is_atom(view) and is_binary(template) or is_atom(template) do
+    conn
+    |> put_view(view)
+    |> render(template, assigns)
+  end
+
+  defp do_render(conn, template, format, assigns)
+    when is_binary(template) and is_binary(format) do
```
Comment (chrismccord) [inline lib/phoenix/controller.ex]: There is only one `do_render` clause so this guard is unnecessary

Comment (scrogson) [inline lib/phoenix/controller.ex]: Good point. Fixed.

File: lib/phoenix/controller.ex [hunk 2]
```diff
@@ -468,26 +472,40 @@ defmodule Phoenix.Controller do
   `layout_formats/2` and `put_layout_formats/2` can be used to configure
   which formats support/require layout rendering (defaults to "html" only).
   """
-  @spec render(Plug.Conn.t, binary | atom, Dict.t) :: Plug.Conn.t
-  def render(conn, template, assigns) when is_atom(template) do
+  @spec render(Plug.Conn.t, binary | atom, atom | binary | Dict.t) :: Plug.Conn.t
```
Comment (josevalim) [inline lib/phoenix/controller.ex]: Wouldn't it be better to define multiple clauses?

```elixir
@spec render(Plug.Conn.t, binary | atom, Dict.t) :: Plug.Conn.t
@spec render(Plug.Conn.t, module, binary | atom) :: Plug.Conn.t
```

?

File: lib/phoenix/controller.ex [hunk 3]
```diff
@@ -468,26 +472,40 @@ defmodule Phoenix.Controller do
...
   end
 
-  def render(conn, template, format, assigns) do
+  def render(conn, view, template)
+    when is_atom(view) and is_binary(template) or is_atom(template) do
+    render(conn, view, template, [])
+  end
+
+  @spec render(Plug.Conn.t, atom, atom | binary, Dict.t) :: Plug.Conn.t
+  def render(conn, view, template, assigns)
```
Comment (josevalim) [inline lib/phoenix/controller.ex]: We should rather make this call the canonical one and have the other variants calling here so we avoid the put_view indirection!

**Comment by josevalim** [comment]
Once we are done, we just need to squash and I think we are good to go!

File: lib/phoenix/controller.ex (hunk 2 above)
Comment (scrogson) [inline lib/phoenix/controller.ex]: I wasn't aware that was valid.

File: lib/phoenix/controller.ex (hunk 3 above)
Comment (josevalim) [inline lib/phoenix/controller.ex]: Please ignore this comment.

**Comment by scrogson** [comment]
Squashed and merged manually cd1fc7c

**Pull request closed without merging**